### ISSUE TITLE
Student's work lost

### ISSUE THREAD

**Issue by psuci** [RunestoneInteractive/RunestoneServer#2028]
First I lost the work then the student did.
All the assignments are blank on her's and mine.
After I lost her assignments I looked at all the possible reasons listed on the Runestone site (wrong course, etc). Then we cleared both our caches (which did not solve it). (Her work was still there on her Ipad at that point).

What is very strange is that all her work disappeared  from her station (Mac Ipad) about a week after I lost her work on my station (MacBook). I know this sounds impossible but I don't think we made an obvious blunder).

Name: Telyna Bacon
UN: Telyna2

**Comment by bnmnetp** [comment]
When you say that all the assignments are blank, I don't really know what you mean and it is important to be precise when trying to figure this out.  

I see more than 1000 records in the log for this student as well as 240 code entries for activecodes.

You see I can count the places in our code where we delete things from the database on one hand.

1. when a user requests deletion of their username.
2. when we age out old data -- that is records > 2 years old

We use an extremely reliable database - Postgresql - it doesn't just randomly lose data.

Your course has a start date of 2020-01-01 ??  Is there a reason you don't create a new course for each term?

Can you give me a particular assignment? 

Did you look in the Javascript Console on your browser to see if you get an error?

What I do find odd is that for your particular course I am not seeing answers to multiple choice questions even though there are 100's of them in the book.  There should be a record of them as your students read the book and answer those questions as part of the text yet I only find 13 answers!!  That is totally unlike any other course I have seen, is there anything you can tell me about how your course is organized that might shed some light?

Does your school have particularly strict firewall rules?  Do all of your students use iPads most of the time?

I'm sure you and your student are frustrated but there is nothing obviously wrong with the system -- it is working fine for many thousands of students -- so Its going to take some help from you to figure out what is going on in your particular situation.

**Comment by bnmnetp** [comment]
looking more closely at your class it seems that this student has TWO usernames Telyna2 and Telyna22

I'm guessing that may be the problem

**Comment by psuci** [comment]
First, this is no tragedy as long as all the students don't start
losing material, because my grade book is caught up and she got credit for
this assignment. So don't spend too much time on it unless you think it
means there is something really wrong.

We have not been using the book. I created all the assignments.

This assignment is one I created and it required them to write some Python
code (50-100 lines). I had to help them extensively so I essentially wrote
the code with this girl and we got it to work. "Blank" means that there is
no code there now. The bar is all the way to the left and it doesn't
respond to move to the right.

Maybe she did something screwy that she didn't tell me about. But in this
case it seems very unlikely since she is getting an A and also the program
was working and she was going to use it to make some cool star quilt
patterns.

If I remember correctly, the code for none of her past assignments is there
(about 20 of them).

Before she lost her code I was unable to see her code on my platform. And I
am sure I did not make any mistakes in that case (like only not paying
attention to the due date, etc.) and,as I said, we cleared both caches. As
it turns out, the admin uses the students cache as a way of monitoring
their activity, so they will not let the students clear it routinely.

I am only concerned that this will happen to other students and then it
might be bad if my grade book is not caught up. So if it seems like some
sort of error on our part, just let it go. I will let you know if it
happens again.

I will ask about the firewall. It seems quite possible that it is a
security issue over here, but then I don't understand why it just happened
to Telyna.

**Comment by bnmnetp** [comment]
Looking at where the code is stored I find the following for Telyna22
```
+------------------------------+-------+
| acid                         | count |
|------------------------------+-------|
| star_quilt_vary_parameters   | 2     |
| star_quilt_vary_parameters_2 | 3     |
+------------------------------+-------+
```

However for Telyna2 I find many more - 66 different coding projects but nothing for the star_quilt .

**Comment by psuci** [comment]
I registered Telyna again under Telyna22 after I found that her coding was
missing.
After the code for "star quilt vary parameters "disappeared" I checked the
Telyna2 assignment called "practice print" and there was nothing there on
her Ipad. Since that was one of the first assignments we did I assumed that
everything was missing on her Ipad. I still have code appearing for that
assignment (and others) at my station.

The code for the assignment "star quilt vary parameters" is missing but I
do have code for "star quilt simple star'.

Let's forget it for now because it is too complicated to untangle. I can't
always be sure the students are following my instructions and maybe somehow
she mixed something up. I will let you know if any more issues like this
come up.

We cannot clear the cache on their Ipads routinely due to security issues.
